src/components/CookieApprovalDialog.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import Cookies from 'js-cookie';
+import Button from './Button';
+import styles from './CookieApprovalDialog.module.scss';
+
+const gdprConsentCookieName = 'newrelic-gdpr-consent';
+
+const CookieApprovalDialog = ({ className, setCookieConsent }) => {
+  const [isCookieSet, setIsCookieSet] = useState(true);
+
+  useEffect(() => {
+    setIsCookieSet(Cookies.get(gdprConsentCookieName) !== undefined);
+  }, []);
+
+  function writeCookie(answer) {
+    const currentEnvironment =
+      process.env.ENV || process.env.NODE_ENV || 'development';
+    const options = { expires: 365 /* days */ };
+    if (currentEnvironment !== 'development') {
+      options.domain = 'newrelic.com';
+    }
+
+    Cookies.set(gdprConsentCookieName, String(!!answer), options);
+    setIsCookieSet(true);
+    answer && setCookieConsent(true);
+  }
+
+  return (
+    !isCookieSet && (
```

Comment:
We may want to just do something like `if (isCookieSet) return null;` above this return statement. That way, we cut down on one more indent level.